src/components/common/YourCompany/YourCompany.vue

```diff
@@ -405,12 +444,17 @@ export default class YourCompany extends Mixins(
     return 'Unknown'
   }
 
+  /** The start date for a firm. */
+  private get businessStartDate (): string {
+    return 'Unknown'
+  }
+
   /** Compare names. */
   private get isNewName () {
     const correctedName = this.getApprovedName
     const currentName = this.isCorrectionFiling
-      ? this.getOriginalIA.incorporationApplication.nameRequest.legalName
-      : this.getBusinessSnapshot.businessInfo.legalName
+      ? this.getOriginalIA?.incorporationApplication?.nameRequest.legalName
+      : this.getBusinessSnapshot?.businessInfo?.legalName
```

Comment:
Why might getOriginalIA or getBusinessSnapshot be falsey?

The risk with using too much conditional chaining is that it hides cases where something might incorrectly be null/undefined. (Like a startup timing issue.)

Reply:
Also -- and you might be watching this already -- I think the business object and NR object are changing a bit, such that some fields have moved.

For example (for registrations at least), legal type is now in the NR object: https://github.com/bcgov/business-create-ui/blob/7404fcbbccccced5736ad316294daa86b80153a9/src/mixins/filing-template-mixin.ts#L333

I _think_ Vysakh_ championed this. But I'm not sure about anything (sorry).

Reply:
For sure, i agree. This was more or less to produce the base of the new flow without throwing a bunch of type errors. 
This needs to be refactor as we work on the individual components and adapt them to the new entity types. 

Will try and preserve the `getBusinessSnapshot` so that it can be either firm or company, so there would both could never be `falsey`.

Reply:
Either way, removed the optional chaining so i don't forget later 👍

Reply:
ok for now